src/main/java/org/rumbledb/server/RumbleHandler.java

```diff
@@ -0,0 +1,203 @@
+package org.rumbledb.server;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.spark.SparkException;
+import org.rumbledb.api.Item;
+import org.rumbledb.cli.JsoniqQueryExecutor;
+import org.rumbledb.config.RumbleRuntimeConfiguration;
+import org.rumbledb.exceptions.CliException;
+import org.rumbledb.exceptions.OurBadException;
+import org.rumbledb.exceptions.RumbleException;
+import org.rumbledb.items.ItemFactory;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import sparksoniq.spark.SparkSessionManager;
+
+@SuppressWarnings("restriction")
+public class RumbleHandler implements HttpHandler {
```

Comment:
I think I would strongly prefer RumbleHttpHandler as a name for this class. RumbleHandler feels very vague

Reply:
DONE